library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/recorder/RecorderOnDrawListener.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.recorder
+
+import android.app.Activity
+import android.content.res.Configuration
+import android.os.Handler
+import android.os.Looper
+import android.view.View
+import android.view.ViewTreeObserver
+import com.datadog.android.sessionreplay.processor.Processor
+import java.lang.ref.WeakReference
+
+internal class RecorderOnDrawListener(
+    activity: Activity,
+    private val pixelsDensity: Float,
+    private val processor: Processor,
+    private val snapshotProducer: SnapshotProducer = SnapshotProducer(),
+    private val handler: Handler = Handler(Looper.getMainLooper())
+) : ViewTreeObserver
+    .OnDrawListener {
+    private var currentOrientation = Configuration.ORIENTATION_UNDEFINED
+    private val trackedActivity: WeakReference<Activity> = WeakReference(activity)
```

Comment:
why do we need to use `WeakReference` if we know activity lifecycle and can just release `trackedActivity` property when activity goes into a destroyed state?

Reply:
Yes, I know but I want to be better safe than sorry. I want to totally avoid keeping those references and it will not be too costly to just use WeakRefs

Reply:
The problem with weak references is that they can be collected at some point and everything will stop working all of the sudden, recording will just stop, it is very hard to debug such things.

Reply:
The way I know is that they will never be destroyed if they still have a strong reference from somewhere in the process. @xgouchet I really want also your input here, should we keep WeakRefs or just go with direct reference and hope that will never go into trouble.

Reply:
ok, I refreshed my knowledge (didn't use `WeakReference` for a long time) and that is true: they are cleared only when referent is weakly reachable.

So I guess we can keep them. But still, I'm curious why do we need it when we have a complete access to the lifecycle.

Reply:
Yup, we can debate this, waiting for @xgouchet's opinion also.